### ISSUE TITLE
Consider allowing some annotations on Builder class

### ISSUE THREAD

**Issue by lombokissues** [projectlombok/lombok#731]
_Migrated from Google Code (issue 696)_

**Comment by simplysoft** [comment]
Adding support for `@ToString(includeFieldNames=false)` to Builder would be great

**Comment by mthmulders** [comment]
Something I often see is a POJO (say `MyPojo`) with `@Builder` on it, and then:

```java
@com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
MyPojoBuilder {
}
```

This way, Jackson can also use the generated `MyPojoBuilder ` and `MyPojo` is truly immutable. Although it works great, it's a bit repetitive. So a `onBuilderClass` attribute for `@Builder` would be nice, allowing you to specify any annotations that must be put on the generated builder class. I suppose it would work pretty much just like `onConstructor` for `@AllArgsConstructor`.

**Comment by mthmulders** [comment]
@rzwitserloot, would you consider a PR that adds `onBuilderClass` to `@Builder`? I have a working concept, just need to polish a few things and add some documentation.

**Comment by rzwitserloot** [comment]
@mthmulders Yeah, that sounds like a good addition. I'm leaving this open for the `onBuilderClass` request.

Let's not allow `@Setter` and co. on Builder. Don't use em as beans. I can't foresee the impact having to support that idea in future updates is going to have.

**Comment by Maaartinus** [comment]
@mthmulders Is `onBuilderClass` a good idea when we can add an empty `public static class Builder {}` and annotate it? It's more readable and doesn't depend on the `@__` hack.

**Comment by mthmulders** [comment]
@Maaartinus It saves you boiler plate code, and as far as I'm concerned, that's one of the main features of project Lombok. Explicitly writing the `XBuilder` class for the sole reason to put an annotation on it feels redundant to me.

**Comment by Maaartinus** [comment]
@mthmulders Sure, but there may be other reasons why you need to write the class (making it package-protected or adding some other stuff). With this annotation parameter, you can save about ten characters, but when you later need to create the builder class, you actually wasted about ten characters. I'm not really opposed to the idea, it's just that it gives me a choice I don't like.

**Comment by mthmulders** [comment]
Maybe I don't understand your concern correctly, @Maaartinus, but if it's a choice you don't like then you're free to pick the other option? You could still write your `XBuilder` by hand and modify it the way you like?

**Comment by Maaartinus** [comment]
@mthmulders I'd prefer not to be able to pick (unless the options bring a lot of value). I hate [TMTOWTDI][1], but it may be just me. Just don't let you stop by it.

[1]: https://en.wikipedia.org/wiki/There's_more_than_one_way_to_do_it

**Comment by mthmulders** [comment]
Update: I'm not planning to work on the `onBuilderClass` annotation. It turns out that #1180 actually saves most of the Jackson-related boilerplate: manually defining and annotating it with `@JsonPOJOBuilder`.
